gae_discourse_client/categories.py

```diff
@@ -0,0 +1,77 @@
+"""Gateway for accessing the Discourse API (for forums)"""
+
+import json
+import re
+from urllib import urlencode
+
+from google.appengine.api import urlfetch
+from google.appengine.ext import ndb
+
+
+class Error(Exception):
+    pass
+
+
+class CategoryClient(object):
+    """An API client for interacting with Discourse for category-related actions"""
+
+    def __init__(self, api_client):
+        self._api_client = api_client
+
+    @ndb.tasklet
+    def getByName(self, category_name, parent_category_name=None):
```

Comment:
General thought: for libraries that are being published, it's a good idea to spend more time on the docs than you normally would. I'd like to see docstrings for all the functions in these client classes, particularly documenting the args and return values. Use the Google formatting style, described [here](http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html).

The ultimate goal for a Python library would be to get Python documentation compiled with Sphinx and uploaded to http://readthedocs.org, as described [here](http://python-guide.readthedocs.org/en/latest/writing/documentation/), but I think all that's overkill for now unless you really want to learn how to do it.